src/JMS/Serializer/GraphNavigator.php

```diff
@@ -210,7 +210,13 @@ public function accept($data, array $type = null, Context $context)
                 }
 
                 if ($context instanceof DeserializationContext && ! empty($metadata->discriminatorMap) && $type['name'] === $metadata->discriminatorBaseClass) {
-                    $metadata = $this->resolveMetadata($data, $metadata);
+                    try {
+                        $metadata = $this->resolveMetadata($data, $metadata);
+                    } catch(\LogicException $e) {
+                        if($metadata->discriminatorStrictDeserialize) {
```

Comment:
$metadata is based on the definition on the root class where the descriminator is defined. So ->resolveMetadata() does simply enrich the Metadata.

Yes, your right. It would make more sense to improve ->resolveMetadata()